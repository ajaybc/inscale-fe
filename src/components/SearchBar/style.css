.SearchBar {
  display: flex;
  justify-content: space-between;
}

.SearchBar__left .react-datepicker__input-container > input {
  padding: 8px;
  width: 250px;
}
.SearchBar__left .react-datepicker__input-container .react-datepicker__close-icon::after {
  background: #1ba94c;
}

.SearchBar__left > div {
  display: inline-block;
  padding-right: 10px;
}

.SearchBar__searchField {
  padding: 8px;
  width: 250px;
}

.SearchBar__searchBtn {
  background: #1ba94c;
  border-color: #1ba94c;
  color: #FFF;
  padding: 8px;
}

.SearchBar__searchBtn:focus {
  outline: none;
}