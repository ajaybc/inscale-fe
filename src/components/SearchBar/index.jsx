import React, {Component} from 'react';
import DatePicker from 'react-datepicker';

import 'react-datepicker/dist/react-datepicker.css';
import './style.css';

export default class SearchBar extends Component {
  render () {
    return <div className="SearchBar">
      <div className="SearchBar__left">
        <DatePicker
          className="test"
          // selected={new Date()}
          onChange={(a, b) => {
            console.log(a,b);
          }}
          placeholderText="Start-Date" 
        />
        <DatePicker
          // selected={new Date()}
          onChange={(a, b) => {
            console.log(a, b);
          }}
          placeholderText="End-Date"
        />
      </div>
      <div className="SearchBar__right">
        <input type="text" className="SearchBar__searchField"/>
        <button className="SearchBar__searchBtn">Search</button>
      </div>
    </div>;
  }
}