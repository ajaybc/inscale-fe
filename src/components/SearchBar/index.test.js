import React from 'react';
import renderer from 'react-test-renderer';
import moment from 'moment';

import SearchBar from './index.jsx';

it('SearchBar renders correctly when filters are provided', () => {
  const tree = renderer
    .create(<SearchBar onSearch={_ => _} startDate={moment('2018-05-01')} endDate={moment('2018-05-05')} query={'test'} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});