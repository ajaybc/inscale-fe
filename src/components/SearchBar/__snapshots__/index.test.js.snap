// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders correctly when filters are provided 1`] = `
<div
  className="SearchBar"
>
  <div
    className="SearchBar__left"
  >
    <div>
      <div
        className="react-datepicker-wrapper"
      >
        <div
          className="react-datepicker__input-container"
        >
          <input
            autoComplete={undefined}
            autoFocus={undefined}
            className="test"
            disabled={false}
            id={undefined}
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Start-Date"
            readOnly={undefined}
            required={undefined}
            tabIndex={undefined}
            title={undefined}
            type="text"
            value="01/05/2018"
          />
          <button
            className="react-datepicker__close-icon"
            onClick={[Function]}
            tabIndex={-1}
            title={undefined}
            type="button"
          />
        </div>
      </div>
    </div>
    <div>
      <div
        className="react-datepicker-wrapper"
      >
        <div
          className="react-datepicker__input-container"
        >
          <input
            autoComplete={undefined}
            autoFocus={undefined}
            className=""
            disabled={false}
            id={undefined}
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="End-Date"
            readOnly={undefined}
            required={undefined}
            tabIndex={undefined}
            title={undefined}
            type="text"
            value="05/05/2018"
          />
          <button
            className="react-datepicker__close-icon"
            onClick={[Function]}
            tabIndex={-1}
            title={undefined}
            type="button"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="SearchBar__right"
  >
    <input
      className="SearchBar__searchField"
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      className="SearchBar__searchBtn"
      onClick={[Function]}
    >
      Search
    </button>
  </div>
</div>
`;
